tasks/backfill_gh_app_installations.py

```diff
@@ -0,0 +1,149 @@
+from services.owner import get_owner_provider_service
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class BackfillGHAppInstallationsTask(
+    BaseCodecovTask, name=backfill_gh_app_installations
+):
+    # Looping and adding all repositories in the installation app
+    async def add_repos_service_ids_from_provider(
+        self,
+        db_session: Session,
+        ownerid: int,
+        owner_service: torngit.base.TorngitBaseAdapter,
+        gh_app_installation: GithubAppInstallation,
+    ):
+        repos = await owner_service.list_repos_using_installation()
+        if repos:
+            # Fetching all repos service ids we have for that owner in the DB
+            repo_service_ids_in_db = [
+                repo.service_id
+                for repo in db_session.query(Repository.service_id)
+                .filter_by(ownerid=ownerid)
+                .all()
+            ]
+
+            # Add service ids from provider that we have DB records for to a list
+            new_repo_service_ids = set()
+            current_repo_service_ids = set(
+                gh_app_installation.repository_service_ids or []
+            )
+            for repo in repos:
+                repo_data = repo["repo"]
+                service_id = repo_data["service_id"]
+                if service_id and service_id in repo_service_ids_in_db:
+                    new_repo_service_ids.add(service_id)
+            log.info(
+                "Added the following repo service ids to this gh app installation",
+                extra=dict(
+                    ownerid=ownerid,
+                    installation_id=gh_app_installation.installation_id,
+                    new_repo_service_ids=list(new_repo_service_ids),
+                ),
+            )
+            gh_app_installation.repository_service_ids = list(
+                current_repo_service_ids.union(new_repo_service_ids)
+            )
+            db_session.commit()
+
+    async def run_async(
+        self,
+        db_session: Session,
+        owner_ids: Optional[List[int]] = None,
+        *args,
+        **kwargs
+    ):
+        # Get all owners we're interested in
+        if owner_ids:
+            owners: List[Owner] = (
+                db_session.query(Owner).filter(Owner.ownerid.in_(owner_ids)).all()
+            )
+        else:
+            owners: List[Owner] = (
+                db_session.query(Owner).filter_by(service="github").all()
+            )
+
+        # Loop through all owner_ids
+        for owner in owners:
+            ownerid = owner.ownerid
+
+            # Check if owner any sort of integration
+            if not owner.integration_id:
+                log.info("No installation work needed", extra=dict(ownerid=ownerid))
+                continue
+
+            # Check if owner has a gh app installation entry
+            gh_app_installation: GithubAppInstallation = (
```

Comment:
Perhaps pull this out of the loop and query for all in the list of owners you are running for.